receiver/hostmetricsreceiver/internal/scraper/processscraper/internal/wmiprocinfo/manager_windows.go

```diff
@@ -0,0 +1,101 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build windows
+
+package wmiprocinfo // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/hostmetricsreceiver/internal/scraper/processscraper/internal/wmiprocinfo"
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/yusufpapurcu/wmi"
+)
+
+func NewManager() Manager {
+	return &wmiProcInfoManager{queryer: wmiProcessQueryer{}}
+}
+
+var (
+	ErrNoProcesses     = errors.New("no process info is currently registered")
+	ErrProcessNotFound = errors.New("process info not found")
+)
+
+type wmiProcInfo struct {
+	handleCount uint32
+	ppid        int64
+}
+
+type wmiQueryer interface {
+	wmiProcessQuery() (map[int64]*wmiProcInfo, error)
+}
+
+type wmiProcInfoManager struct {
+	queryer   wmiQueryer
+	processes map[int64]*wmiProcInfo
+}
+
+func (m *wmiProcInfoManager) Refresh() error {
+	processes, err := m.queryer.wmiProcessQuery()
+	if err != nil {
+		return err
+	}
+	m.processes = processes
+	return nil
+}
+
+func (m *wmiProcInfoManager) getProcessInfo(pid int64) (*wmiProcInfo, error) {
+	if len(m.processes) == 0 {
+		return nil, ErrNoProcesses
+	}
+	procInfo, ok := m.processes[pid]
+	if !ok {
+		return nil, fmt.Errorf("%w for %d", ErrProcessNotFound, pid)
+	}
+	return procInfo, nil
+}
+
+func (m *wmiProcInfoManager) GetProcessHandleCount(pid int64) (uint32, error) {
+	procInfo, err := m.getProcessInfo(pid)
+	if err != nil {
+		return 0, err
+	}
+	return procInfo.handleCount, nil
+}
+
+func (m *wmiProcInfoManager) GetProcessPpid(pid int64) (int64, error) {
+	procInfo, err := m.getProcessInfo(pid)
+	if err != nil {
+		return 0, err
+	}
+	return procInfo.ppid, nil
+}
+
+type wmiProcessQueryer struct{}
+
+//revive:disable-next-line:var-naming
+type Win32_Process struct {
+	ProcessID       int64
+	ParentProcessID int64
+	HandleCount     uint32
+}
+
+func (wmiProcessQueryer) wmiProcessQuery() (map[int64]*wmiProcInfo, error) {
+	processes := []Win32_Process{}
+	// Based on reflection of Win32_Process type, this creates the following query:
+	// `get-wmiobject -query "select ProcessId, ParentProcessId, HandleCount from Win32_Process"`
```

Comment:
This method retrieves both parent process ID and process handle count on every run, regardless of whether both are needed.

Would it make sense to call different commands depending on which information is needed? Would it help in performance to for example only retrieve parent process ID and not retrieve handle count, assuming the user didn't enable the `process.handles` metric?

The default configuration for the process scraper is to require parent process ID (as the `process.parent_pid` resource attribute is enabled by default) and not require handle count (as the `process.handles` metric is disabled by default).